### PR TITLE
Added Row to BsonDocument conversion

### PR DESCRIPTION
**Pull request by rozza**
SPARK-310

### CODE REVIEW

**Comment by rozza** [comment]
Its rewritten code from the original scala version.

File: src/main/java/com/mongodb/spark/sql/connector/schema/ConverterHelper.java
Comment (stIncMale) [inline src/main/java/com/mongodb/spark/sql/connector/schema/ConverterHelper.java:50]: We should convert `ConverterHelper` to a `final` class with a private constructor for the following reasons:

1. The ability to define `static` methods in interfaces was added to Java because it [_"makes it easier for you to organize helper methods in your libraries; you can keep static methods specific to an interface in the same interface rather than in a separate class"_](https://docs.oracle.com/javase/tutorial/java/IandI/defaultmethods.html#static). However, this is not applicable to the interface `ConverterHelper`: the interface is meant to be solely a collection of `static` methods and is not meant to define a type.
2. The purpose of `static` methods `getBsonValueCodec`/`getDefaultJsonWriterSettings` appear to be to provide an identity-agnostic way of getting a BSON value codec / default JSON writer settings. However, the fact that the `static` fields `BSON_VALUE_CODEC`/`DEFAULT_JSON_WRITER_SETTINGS` are also `public`, diminishes the benefit of these methods, because a user can still access the BSON value codec / default JSON writer settings in a way that then allows to rely on their identity.

Generally speaking, an interface may be used only if it is meant to define a type, i.e., to be implemented.

File: src/test/java/com/mongodb/spark/sql/connector/schema/RowToBsonDocumentConverterTest.java
Comment (stIncMale) [inline src/test/java/com/mongodb/spark/sql/connector/schema/RowToBsonDocumentConverterTest.java:210]: It seems, this method is not used and can be removed.

File: src/main/java/com/mongodb/spark/sql/connector/schema/RowToBsonDocumentConverter.java
Comment (stIncMale) [inline src/main/java/com/mongodb/spark/sql/connector/schema/RowToBsonDocumentConverter.java:53]: I suggest changing this interface to a class for mostly the same reasons as in this [comment](https://github.com/mongodb/mongo-spark/pull/54/files#r716296848).

**Review by stIncMale — CHANGES_REQUESTED** [review]

File: src/main/java/com/mongodb/spark/sql/connector/schema/RowToBsonDocumentConverter.java
Comment (rozza) [reply src/main/java/com/mongodb/spark/sql/connector/schema/RowToBsonDocumentConverter.java:53]: Done

File: src/main/java/com/mongodb/spark/sql/connector/schema/ConverterHelper.java
Comment (rozza) [reply src/main/java/com/mongodb/spark/sql/connector/schema/ConverterHelper.java]: I like it better as you suggested - thanks @stIncMale

File: src/main/java/com/mongodb/spark/sql/connector/schema/RowToBsonDocumentConverter.java
Comment (stIncMale) [reply src/main/java/com/mongodb/spark/sql/connector/schema/RowToBsonDocumentConverter.java:53]: I do not see related changes in the second commit, maybe they were not pushed.

But the above observation is good:), because I also noticed that `BsonDocumentToRowConverter` is an instantiable class, while `RowToBsonDocumentConverter` is a collection of static methods. I understand that `RowToBsonDocumentConverter` does not need to be instantiable, but the inconsistency in what a converter is (an object or a `static` method) is not good. I would like to change the [previous proposal](https://github.com/mongodb/mongo-spark/pull/54#discussion_r716311756) and propose to consider changing this interface not to a class with `static` methods, but to an instantiable class for consistency with `BsonDocumentToRowConverter`.

**Comment by jyemin** [comment]
@rozza I'm going to leave this to @stIncMale so feel free to merge after he approves.

File: src/main/java/com/mongodb/spark/sql/connector/schema/RowToBsonDocumentConverter.java
Comment (rozza) [reply src/main/java/com/mongodb/spark/sql/connector/schema/RowToBsonDocumentConverter.java]: A happy accident! I've made both converters final classes. Thanks @stIncMale

**Review by stIncMale — APPROVED** [review]

**Comment by jyemin** [comment]
@rozza one LGTM is sufficient for this one.

**Pull request merged**
